src/store/styleguide/property/object-property.ts

```diff
@@ -58,16 +62,29 @@ export class ObjectProperty extends Property {
 		return 'object';
 	}
 
+	private resolveProperties(): Map<string, Property> {
+		if (!this._properties) {
+			if (!this.propertyResolver) {
+				throw new Error('property resolver is not set');
+			}
+
+			const properties = this.propertyResolver();
+			this._properties = new Map();
+
+			// tslint:disable-next-line:no-non-null-assertion
+			properties.forEach(property => this._properties!.set(property.getId(), property));
```

Comment:
It is a bit strange that you have to use a `!` here.
Could we not just make the `_properties` non optional and have it always be a empty Map.
Then you would not have to disable tslint for this line :)

Reply:
I guess @faselbaum is trying to use "undefined" as a lazy indicator.
Alternative would be to use another const and assign it at the end of the method. This would also ensure that the data structure is not half-constructed if there is a problem.